### ISSUE TITLE
Oktoberfest Ad blocks biometric unlock

### ISSUE THREAD

**Issue by bluezed** [PhilippC/keepass2android#1412]
I just wanted to log into a website and opened Keepass2Android on my phone. There was a Oktoberfest Ad asking for a donation and in front of it an empty biometric unlock dialog. See screenshot.

It was not possible to dismiss this dialog and I was forced to restart the entire phone.

![Screenshot_2020-09-19-11-23-25-492_keepass2android keepass2android](https://user-images.githubusercontent.com/3831765/93663999-ecd44380-fa6b-11ea-976a-0498316f1491.jpg)

Device: Xiaomi Mi Note 10 Lite
Android: 10
Keepass2Android from PlayStore v 1.08b-r1

**Comment by chris42** [comment]
That is the developer asking if you Like Keepass2Android and since Oktoberfest has started, if you want to sponsor him a beer. So no ad for Oktoberfest, more for Keepass2Android itself.

The biometric part is the part that is used to unlock your database.

Are you a first time user? Then probably @PhilippC this should be in English. 
If you are not a first time user, do you have your GUI in English? Then this might be a translation error.
If you are not a first time user and use a German GUI without understanding it, you should check your life choices. ;-)

**Comment by bluezed** [comment]
I'm a long time user and my UI is in German.

I'm not objecting to the developer asking for donations but I'm objecting to the fact that I had to restart my phone because of this.

**Comment by chris42** [comment]
I see that I did not understand your description correctly.

On my phone (password protected) I can just click "So gut gefällt es mir dann auch nicht" then I get the password prompt. I assume that was not possible because of the biometric check you have on your phone?

That would then be indeed a bug.

**Comment by bluezed** [comment]
Yes, that's exactly it. The biometric unlock in the foreground is blocking the phone and it does not accept fingerprint or "Abbrechen". It just doesn't react.

The navigation buttons at the bottom and hardware keys on the phone still work (that's how I got the screenshot) but the biometric dialog never disappears. Only a phone restart helped to regain control.

**Comment by chris42** [comment]
Then @PhilippC probably needs to look into this, as it sound like a bug in the GUI. 
For Password protected phones, I just tried it and it works without issues as describes above.

**Comment by magne-hov** [comment]
Same thing just happened to me, Oktoberfest prompt covered by biometric pane. Unable to unlock with fingerprint, and soft navigation gestures were only able to get me to homescreen with biometric pane still blocking me from doing anything else. Had to restart phone.

Deivce: Oneplus 5T
Model: ONEPLUS A5010
Build: ONEPLUS A5010_43_200224
Android: 9
OxygenOS version: 9.0.11
Keepass2Android from PlayStore v 1.08b-r1

**Comment by PhilippC** [comment]
Hi @bluezed and @magne-hov ,
sorry for the delay in looking into this. Oktoberfest is now over, but the next donation requests will come next year so I'd like to fix this annoyance. Unfortunately I cannot reproduce. On my device, if I change the date manually to May 12th in some future year (eg 2021), I see the "it's my birthday..." message and a little toast saying "Fingerprint unlock canceled" which is to be expected as the app should cancel the fingerprint request when the unlock screen goes into background.

Can you please describe in more detail what you did to get this behavior? I tried with both QuickUnlock and Full unlock. Maybe it's related to some other stuff? Can you reproduce if you change the date of your phone manually to May 12th 2021?

**Comment by bluezed** [comment]
@PhilippC thanks for looking into this.
I just tried with setting the date to 12th May 2021 and it behaved as you described. I cannot reproduce the issue like this but it happened twice with the Oktoberfest ad when it showed up. In both cases it was with QuickUnlock.

What I did was... I was using Fennec F-Droid (v68) to browse to a website where I wanted to log in. I tapped on the "Mit Keepass2Android ausfüllen" prompt. Keepass opened with the Oktoberfest ad and the dead biometric unlock dialog.

I did have a similar issue with an app called Cryptomator which sometimes displayed the dialog twice, only the second one was working and the first one blocked the phone. Apparently their attempt to dismiss the biometric prompt did not always work and it seems that the issue here is along the same lines...
They managed to put a workaround in place but not sure how it works really... here's the link to the forum thread where I reported the issue: [Cryptomator Forum](https://community.cryptomator.org/t/biometric-unlock-opens-twice-and-blocks-entire-phone/6377)

Hope this is useful?
